Provide faraday-telemetry command line functionality

kb1lqc opened the pull request:
This PR provides the code necessary to:
* Install faraday-telemetry while performing a `pip install faraday`. Currently only usable in the editable local mode since at the time of the PR creation no new PyPi packages have been uploaded.
* Configure faraday-telemetry via command line
* Command certain functionality such as log saving and deletion via command line

Overall this PR is to get this code exposure. It needs to be cleaned up a tad but is 98% there. Some functionality has not been implemented but was captured in #209. Also, any package built from this PR will provide `faraday-proxy` and `faraday-telemetry` after the pip installation.

kb1lqc committed:
Added faraday-telemetry to command setuptools

Added faraday-telemetry command to setuptools. Can start telemetry as
preconfigured (manually) from command line after editable install. Need to
store database in permanent location and then provide command line
configuration.

kb1lqc committed:
Moved sample telemetry config

Moved the telemetry.sample.ini into the etc/faraday folder. Then, I added
the etc/faraday location to setup.cfg for setuptools to use.

kb1lqc committed:
Updated telemetry configuration

Load in logging.ini and telemetry.ini during startup of telemetry.py
script. This pulls in a common logging.ini. Also, I realized that the
Proxy and Telemetry sections of logging configs aren't even used so I am
going to delete them in the next commit.

kb1lqc committed:
realized logging.ini operation

Misspelled 'Telemetry' as 'telem' in the logger_Telemetry section which
caused telemetry to default to root configuration. This made me think none
of it worked. In reality by deleting logger_Proxy and logger_telemetry I
broke proxy! All fixed.

kb1lqc committed:
Creating telemetry.db in user folder

Modified proxy initdb function to read in the telemetry.ini configuration
file parameters and create a database in the user .faraday folder. Also
was required to change the telemetry schema db.sql to telemetry.sql to
avoid conflict with proxy application schema.

kb1lqc committed:
Logging to Telemetry Database

Moved in some more proxy sqlinsert function code and modified it for
telemetry. Definitely a bummer this is not more module as I'm definitely
rewriting very similar code. Maybe an upgrade later on.

kb1lqc committed:
Telemetry querydb function working

Updated the configuration file reading code in querydb such that it works
and allows a localhost telemetry query to work!

kb1lqc committed:
Telemetry stations query now works

Fixed the database name query such that the /stations query works.

kb1lqc committed:
Added command line options to telemetry

boilerplate for now, next up is using this initial command line info to do
stuff.

kb1lqc committed:
Updated telemetry.sample.ini

Changed the schemaname to telemetrydb.sql

kb1lqc committed:
Telemetry configuration

Telemetry configuration file can now be updated and initialized via
command line. Still only affecting callsign and nodeid. More to follow.

kb1lqc committed:
Added additional configuration command line op

Added in database and flask server command line options.

kb1lqc committed:
Added number to telemetry section

This allows one to specify which configuration to use if only one unit is
being read.

kb1lqc committed:
Telemetery database actions added

Modified the telemetry.py command line structure to be able to save,
initialize, and show the current log files. Currently shows all .db in the
folder but good enough for now.

kb1lqc committed:
Fixed pytest errors

Updated formatting of telemetry.py to pass our current PEP8 based pytest
checks.

kb1lqd commented:
* Install worked as expected

* ISSUE: If I delete the config file in the telemetry folder and `--init-config` the file never appears again... I don't know where this file is being manipulated.
  * EDIT: Wait the pip install will not manipulate the GITHUB repo I'm looking at... disregard?
![image](https://user-images.githubusercontent.com/339144/27470766-3d7e3178-57aa-11e7-9db4-ae03027795e6.png)

* Running a unit with simple `faraday-telemetry --callsign kb1lqd --nodeid 2` seems to work but it is being saved into a random lib folder....

`faraday-telemetry --callsign kb1lqd --nodeid 2 --database brent.db` produced active telemetry in this folder in C: users .faraday folder. Should we make this more accessible? 

![image](https://user-images.githubusercontent.com/339144/27471278-102a437c-57ac-11e7-9f0e-393fa2636f3c.png)

So I think this is actually a good PR and with further review looks acceptable, unless I'm missing something @kb1lqc

kb1lqc commented:
Haha sorry I totally glossed over how that works.

* Yes saving log files to `~/.faraday/lib` is expected and correct, Proxy does the same.
* The config files will be preferentially located in `/etc/faraday/` in the github repo or in `C:/python27/etc/faraday` and lastly in the local folder but 99.99% of the time it will be in the first two

Thanks for checking!

kb1lqd commented on etc/faraday/telemetry.sample.ini:
```diff
@@ -4,9 +4,10 @@ PORT=8001
 [DATABASE]
 FILENAME=telemetry.db
-SCHEMANAME=db.sql
+SCHEMANAME=telemetrydb.sql
```
Good call on proper naming.

kb1lqd reviewed (approved):
Looks good, no issues noted during code review and updates installed and appeared to properly run and generate telemetry data/files.

kb1lqc committed:
Fixed up some PEP8, Spelling, and formatting issues

Caught that there were comments copied from proxy that still relatd to
proxy instead of telemetry. Also had a few formatting and spacing fixes.
Nothing functional just aesthetics :)

kb1lqc committed:
Updated setup.cfg version

0.0.1011 is Proxy + Telemetry
